kefir.js.flow

```diff
@@ -0,0 +1,160 @@
+/* @flow */
+
+export type Event<T> =
+  {type: 'value', value: T} |
+  {type: 'error', value: any} |
```

Comment:
I've typed the error type, and made it default to being inferred so that it doesn't need to be specified.

I hadn't seen that flow-typed declaration file before. Handy reference to compare to! Here's my notes about comparing theirs to mine. Feel free to read or skip depending on your interest level.

One difference is that theirs tries harder to keep track of properties and streams as different types (I never bothered much since Kefir has the same methods on both anyway).

It has a bunch of overloads depending on number of parameters for `Kefir.merge` and `Kefir.zip`. The Kefir.merge overloads are no longer necessary if you mark Observable's type parameter as covariant, which mine does. (This lets Flow know that it's fine to cast `Observable<Subclass>` to `Observable<Superclass>`. So if `Kefir.merge` is typed to take an array of `Observable<V>`, and you pass it an array of different types of streams, Flow automatically casts all of the streams to the union of all of their types, and the resulting merged stream has that union type.) The Kefir.zip overloads are to work around https://github.com/facebook/flow/issues/1251. (`Kefir.zip` and `Kefir.combine` have a similar type signature to `Promise.all`, but Flow doesn't support defining functions with that exact behavior. Flow's support for Promise.all's type is hard-coded.) The workaround isn't perfect, it only works for up to five parameters and is pretty verbose. (They could have done the same thing for Kefir.combine but didn't.) Might be worth doing.

One interesting issue with the Flow-typed declarations is in the *observable methods* combine and zip:

```js
    combine<U,F,R>(otherObs: Observable<U,F>, combinator?: (a: T, b: U) => R): Stream<R,E|F>;
    zip<U,F,R>(obs: Observable<U,F> | U[], combinator?: (a: T, b: U) => R): Stream<[T,U]|R,E|F>;
```

If combine is called without a combinator function, then it won't know what type the resulting stream is. If zip is called with a combinator function, then Flow will think the stream's values are either of the type returned by the combinator, or an 2-item array of both streams' types. (For example, if the combinator returned a string, you called onValue on the resulting stream, and in the callback tried to use the value as a string, then you'd get a Flow error if your code didn't also support arrays.)

My version defines separate overloads for with and without the combinator function which avoids that issue:

```js
  combine<V2,E2>(otherObs: Observable<V2,E2>): Observable<[V,V2],E|E2>;
  combine<V2,V3,E2>(otherObs: Observable<V2,E2>, combinator: (v: V, v2: V2) => V3): Observable<V3,E|E2>;
  zip<V2,E2>(otherObs: Observable<V2,E2>): Observable<[V,V2],E|E2>;
  zip<V2,V3,E2>(otherObs: Observable<V2,E2>, combinator: (v: V, v2: V2) => V3): Observable<V3,E|E2>;
```